### PR TITLE
Bonfires from Botany

### PR DESCRIPTION
**Pull request by Ergovisavi**
:cl: 
add: Added bonfires to botany. Requires five tower cap logs to create, and mobs can be buckled to them if an iron rod is added to the bonfire and the mob is restrained.
/:cl:

I decided not enough people were burning to death for arbitrary reasons, so here are bonfires.
- You can light things on fire by tossing it into the wood pile. 
- Add an iron rod to buckle people to it for burning witches/wizards more appropriately
- Add natural ambient lighting to your abode without relying on disgusting glowshrooms being everywhere
- Make the AI cry when people keep walking into the bonfire

This might also make Ashwalkers a little more thematic, as you can cook goliath meat on these, if someone wants to bother adding some logs to the ash walker ruin.

bonfire_2

### CODE REVIEW

**Comment by Chocobro** [comment]
sprites are lit af
pun intended
:+1:

**Comment by Jalleo** [comment]
Fire extinguishers can put these out right?

**Comment by Ergovisavi** [comment]
@Jalleo Yes fire extinguishers can extinguish them, as will a lack of oxygen.

**Comment by Aloraydrel** [comment]
shieeet this looks nice

File: code/modules/hydroponics/grown/towercap.dm [hunk 1]
```diff
@@ -90,4 +90,113 @@
...
+	icon = 'icons/obj/hydroponics/equipment.dmi'
+	icon_state = "bonfire"
+	density = FALSE
+	anchored = TRUE
+	buckle_lying = 0
+	var/burning = 0
+	var/flame_strength = FLAMMABLE
+	var/fire_stack_strength = 5
+
+/obj/structure/bonfire/attackby(obj/item/W, mob/user, params)
```
Comment (Jalleo) [inline code/modules/hydroponics/grown/towercap.dm]: Suggestion: add ability to add more logs to the fire so it lasts longer.

**Comment by Niknakflak** [comment]
Remove firepits and replace them with these

**Comment by lzimann** [comment]
:-1: for burning the human and not the lizard

**Comment by Hatfish** [comment]
Now all we need are upright spears to impale heads on and we can go full castlestation.

**Comment by GeneralGoose** [comment]
ligger burning a human and not vice versa not kosher

**Comment by PKPenguin321** [comment]
nice

File: code/modules/hydroponics/grown/towercap.dm [hunk 2]
```diff
@@ -90,4 +90,113 @@
...
+		var/obj/item/stack/rods/R = W
+		R.use(1)
+		can_buckle = 1
+		buckle_requires_restraints = 1
+		user << "<span class='italics'>You add a rod to [src]."
+		var/image/U = image(icon='icons/obj/hydroponics/equipment.dmi',icon_state="bonfire_rod",pixel_y=16)
+		underlays += U
+	if(W.is_hot())
+		StartBurning()
+	return
```
Comment (RemieRichards) [inline code/modules/hydroponics/grown/towercap.dm]: trailing return, remove it please.

File: code/modules/hydroponics/grown/towercap.dm [hunk 3]
```diff
@@ -90,4 +90,113 @@
...
+		user << "<span class='warning'>You need to extinguish [src] before removing the logs!"
+		return
+	if(!buckled_mobs && do_after(user, 50, target = src))
+		var/I
+		for(I=0,I<5,I++)
+			var/obj/item/weapon/grown/log/L = new /obj/item/weapon/grown/log(src.loc)
+			L.pixel_x += rand(1,4)
+			L.pixel_y += rand(1,4)
+		qdel(src)
+		return
```
Comment (RemieRichards) [inline code/modules/hydroponics/grown/towercap.dm]: trailing return, remove it please.

File: code/modules/hydroponics/grown/towercap.dm [hunk 4]
```diff
@@ -90,4 +90,113 @@
...
+	return
```
Comment (RemieRichards) [inline code/modules/hydroponics/grown/towercap.dm]: trailing return, remove it please.

File: code/modules/hydroponics/grown/towercap.dm [hunk 5]
```diff
@@ -90,4 +90,113 @@
...
+		Burn()
+		START_PROCESSING(SSobj, src)
+
+/obj/structure/bonfire/fire_act()
+	StartBurning()
+
+/obj/structure/bonfire/Crossed(atom/movable/AM)
+	if(burning)
+		Burn()
+	return
```
Comment (RemieRichards) [inline code/modules/hydroponics/grown/towercap.dm]: trailing return, remove it please.

File: code/modules/hydroponics/grown/towercap.dm [hunk 6]
```diff
@@ -90,4 +90,113 @@
...
+		if(isobj(A))
+			var/obj/O = A
+			if(O.burn_state < flame_strength)
+				continue
+			O.fire_act()
+		else if(isliving(A))
+			var/mob/living/L = A
+			L.adjust_fire_stacks(fire_stack_strength)
+			L.IgniteMob()
+	return
```
Comment (RemieRichards) [inline code/modules/hydroponics/grown/towercap.dm]: trailing return, remove it please.

File: code/modules/hydroponics/grown/towercap.dm [hunk 7]
```diff
@@ -90,4 +90,113 @@
...
+
+/obj/structure/bonfire/process()
+	if(!CheckOxygen())
+		extinguish()
+		return
+	Burn()
+	return
```
Comment (RemieRichards) [inline code/modules/hydroponics/grown/towercap.dm]: trailing return, remove it please.

File: code/modules/hydroponics/grown/towercap.dm [hunk 8]
```diff
@@ -90,4 +90,113 @@
...
+
+/obj/structure/bonfire/extinguish()
+	if(burning)
+		icon_state = "bonfire"
+		burning = 0
+		SetLuminosity(0)
+		STOP_PROCESSING(SSobj, src)
+	return
```
Comment (RemieRichards) [inline code/modules/hydroponics/grown/towercap.dm]: trailing return, remove it please.

**Comment by Bawhoppen** [comment]
why didn't you just use the already existing burning sprite for the bonfire being lit instead of that weird one, I bet it would fit perfectly in place anyways

**Comment by Ergovisavi** [comment]
Maybe because the existing burning sprite looks like ass and is tiny

**Comment by Bawhoppen** [comment]
except that's not true?

File: code/modules/hydroponics/grown/towercap.dm (hunk 3 above)
Comment (Ergovisavi) [inline code/modules/hydroponics/grown/towercap.dm]: This is no longer a trailing return

**Comment by Alek2ander** [comment]
Stabbing it with a sword should reanimate all corpses as hostile NPC zombies

File: code/modules/hydroponics/grown/towercap.dm [hunk 9]
```diff
@@ -90,4 +90,108 @@
...
+		StartBurning()
+
+
+/obj/structure/bonfire/attack_hand(mob/user)
+	if(burning)
+		user << "<span class='warning'>You need to extinguish [src] before removing the logs!"
+		return
+	if(buckled_mobs.len == 0 && do_after(user, 50, target = src))
+		var/I
+		for(I=0,I<5,I++)
```
Comment (ChangelingRain) [inline code/modules/hydroponics/grown/towercap.dm]: You should make this for into `for(var/I in 1 to 5)` which is both slightly faster and easier to parse at a glance, and will also take up one less line.

**Comment by Ergovisavi** [comment]
We should be good to go here, Dark Souls reference memes aside.

**Pull request merged**